### PR TITLE
Remove 'id' field from infos, compute an id during serialization

### PR DESCRIPTION
**Pull request by sigmundch**

### CODE REVIEW

**Comment by sigmundch** [comment]
@hterkelsen - I took a quick look at your analysis tool and I believe this wont break it, but let me know if you believe that's not the case (in particular, your tool uses the id from the json, but it doesn't appear like you depend on the id from the API), correct?

**Comment by sigmundch** [comment]
@LorenVS - FYI - I don't believe the proto codec would be affected by this.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: dart2js_info
-version: 0.5.17
+version: 0.6.0-dev.0.0
```
Comment (kevmoo) [inline pubspec.yaml]: `0.6.0-dev` is plenty, yo. 😄

**Review by kevmoo — APPROVED** [review]
As long as Ids remain stable 😄

File: pubspec.yaml (hunk 1 above)
Comment (sigmundch) [reply pubspec.yaml]: done :)

File: lib/json_info_codec.dart [hunk 2]
```diff
@@ -5,8 +5,8 @@
 /// Converters and codecs for converting between JSON and [Info] classes.
 part of dart2js_info.info;
 
-List<String> _toSortedSerializIds(Iterable<Info> infos) =>
-    infos.map((i) => i.serializedId).toList()..sort(compareNatural);
+List<String> _toSortedSerializIds(Iterable<Info> infos, Map<Info, Id> ids) =>
```
Comment (harryterkelsen) [inline lib/json_info_codec.dart]: _toSortedSerializIds -> _toSortedSerializedIds

Comment (sigmundch) [reply lib/json_info_codec.dart]: done, thanks

**Pull request merged**